README.md
# EMBER
This is the version of EMBER obtained from server atisha around September 2015 and modified to work with PHP 5 (it was running under PHP 3 at that time) and Chrome.

## Installation

You will need PHP, Apache an MySQL installed - please refer to other guides available on the Internet on how to do that. 

Instructions below are only guidelines and may differ slightly on your system (these refer to Mac OS Yosemite).

```
Make sure you secure your Web server and database with the appropriate measures - e.g. make sure you use HTTPS 
as passwords are passed in plaintext.
```

### Enable PHP for Apache

First, make a backup of the default Apache configuration.

```
# cd /etc/apache2/
# sudo cp httpd.conf httpd.conf.bak
```

Now edit the Apache configuration. Feel free to use TextEdit if you are not familiar with vi.

```
# sudo vi httpd.conf
```

Uncomment the following line:

```
LoadModule php5_module libexec/apache2/libphp5.so
```

Restart Apache:

```
# sudo apachectl restart
```

You can verify PHP is enabled by creating a phpinfo() page in your DocumentRoot (from your Apache configuration - the default DocumentRoot for Mac OS X Yosemite is /Library/WebServer/Documents).

```
# echo '<?php phpinfo();' > /Library/WebServer/Documents/phpinfo.php
```

### MySQL

Either download the MySQL DMG for Mac OS X or install via package manager, e.g. homebrew.

Start MySQL server (you may not need sudo on your system):

```
# sudo mysql.server start
```

Connect to your MySQL console and create database called 'ember'. Create user 'ember' as well and grant all privileges to that user on database 'ember'. Remember to set the password you used for user 'ember' in `html/connect.inc.php` file too.

From command line load the .sql dump file (obtained separately).

```
mysql -u ember -p -h localhost ember < ember-db-dump-2015-09-25.sql
```

### Connect PHP and MySQL
You need to ensure PHP and MySQL can communicate with one another. In the case you not have `/var/mysql/mysql.sock` file after starting MySQL server, it is necessary to create a symlink to the MySQL's sock file (located in `/tmp/mysql.sock` on Mac OS).

```
# cd /var 
# sudo mkdir mysql
# sudo cd mysql
# # ln -s /tmp/mysql.sock mysql.sock
```

Sometimes it is the other way around - you may have `/var/mysql/mysql.sock` and not `/tmp/mysql.sock`, in which case you need to create a symlink `tmp/mysql.sock` to point to `/var/mysql/mysql.sock`.

### Additional configuration 

You may need to configure DocumentRoot in your Apache configuration file (`/etc/apache2/httpd.conf`) to look something like the following:

```
DocumentRoot "/Library/WebServer/Documents"
<Directory "/Library/WebServer/Documents">
    # The Options directive is both complicated and important.  Please see
    # http://httpd.apache.org/docs/2.4/mod/core.html#options
    # for more information.
    #
    Options FollowSymLinks Multiviews
    MultiviewsMatch Any

    #
    # AllowOverride controls what directives may be placed in .htaccess files.
    # It can be "All", "None", or any combination of the keywords:
    #   AllowOverride FileInfo AuthConfig Limit
    #
    AllowOverride None

    #
    # Controls who can get stuff from this server.
    #
    Require all granted
</Directory>
```

```
Note the `Require all granted` directive - the syntax has changed as of Apache 2.4.
```


html/RELEASENOTES
Mon Oct 13 13:49:21 DFT 2003 Rob Buis(ETI)

- updated links for Chapter 2 step 1(BLASTp), Chapter 3 info 3.3.1(Blocks), Chapter 6 Info 6.1.1(techniques).
- removed link on Chapter 3 Info 3.4.1(PRINTS), Chapter 4 step 2(CINEMA), Chapter 6 Info 6.5 (PDB) and
  Chapter 8 Info 8.3 (SAP).
- Made Fig4.7 show up with its true dimension.
- fixed the database connection problems for Fig2 on Chapter 4 Info 4.3 and Fig7 on Chapter 7 Info 7.2.1.


html/newuser.php
<?php
	$go = false;
	$coursenum = $_POST["hiddencourse"];
	include ("connect.inc");
	$result = mysql_query("select * from tblember_course") or die (mysql_error());
	$courseoptions = "<OPTION value=\"0\">Please select...";
	while($row = mysql_fetch_array($result))
	{
		$course = $row["course_id"];
		$courseid = sprintf("EMBER%04d", $course);
		if($course == 1)
			$courseid = "Independent";
		$extra = ($coursenum == $course) ? "SELECTED" : "";
		$courseoptions = $courseoptions."<OPTION $extra value=\"$course\">$courseid";
	}
	mysql_free_result($result);
	if(isset($_POST["hiddenusername"]))
	{
		$user = $_POST["hiddenusername"];
		$passwd = $_POST["hiddenpassword"];
		$firstname = $_POST["hiddenfirstname"];
		$lastname = $_POST["hiddenlastname"];
		$mail = $_POST["hiddenmailaddress"];
		$institution = $_POST["hiddeninstitution"];
		$country = $_POST["hiddencountry"];
		$background = $_POST["hiddenbackground"];
		$qualifications = $_POST["hiddenqualifications"];
		$currentlevel = $_POST["hiddencurrentlevel"];
		$hint = $_POST["hiddenhint"];
		$uniq = $_POST["hiddenunique"];
		$result = mysql_query("select * from tblember_user where user_name = '$user'") or die (mysql_error());
		if(mysql_num_rows($result))
		{
			// duplicate user name was chosen
			$error = "<font color=\"#990000\" /><b>Sorry, the user name $user already exists. Please choose another.</b></font>";
			mysql_free_result($result);
		}
		else
		{
			// check course unique id
			$add = $coursenum == 1; // independent doesnt have to be checked
			if(!$add)
			{
				$result = mysql_query("select * from tblember_course where course_id='$coursenum' AND course_access_code='$uniq'") or die (mysql_error());
				if(!mysql_num_rows($result))
				{
					$error = "<font color=\"#990000\" /><b>Sorry, the entered access code is not correct for course EMBER00$coursenum.</b></font>";
					mysql_free_result($result);
				}
				else
					$add = true;
			}
			if($add)
			{
				// everything ok, add the user
				mysql_query("insert into tblember_user VALUES('$user', '$passwd', '$firstname', '$lastname', '$hint', '$mail', '$coursenum', '$institution', '$country', '', '$background', '$qualifications', '$currentlevel', '0')") or die (mysql_error());
				mysql_free_result($result);
				echo "<script language=\"JavaScript\" type=\"text/JavaScript\" src=\"scripts.js\"></script>";
				echo "<script language=\"JavaScript\">";
				echo "setCookie('username', '$user');";
				echo "document.location.href = \"confirm.php?user=$user\";";
				echo "</script>";
			}
		}
	}
?>
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"><html>
<head>
<title>EMBER: Account</title>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<!-- Fireworks MX Dreamweaver MX target. Created Wed Oct 16 13:14:05 GMT-2200 2002-->
<script language="JavaScript" type="text/JavaScript">
<!-- 

function MM_preloadImages() { //v3.0
 var d=document; if(d.images){ if(!d.MM_p) d.MM_p=new Array();
  var i,j=d.MM_p.length,a=MM_preloadImages.arguments; for(i=0; i<a.length; i++)
  if (a[i].indexOf("#")!=0){ d.MM_p[j]=new Image; d.MM_p[j++].src=a[i];}}
}

//-->
</script>
<script language="JavaScript" type="text/JavaScript" src="scripts.js">
</script>
<link href="screen.css" rel="stylesheet" type="text/css">
</head>
<body bgcolor="#9dafd1" background="Graphics/background_left.jpg" onLoad="MM_preloadImages('Graphics/banner_top_r2_c2_f2.jpg')">
<?PHP
	include "notepad.php";
	?>
<div style="z-index:1; position:absolute; left:0px; top:0px">
<table width="750" border="0" cellpadding="0" cellspacing="0"><!--DWLayoutTable-->
 <tr>
  <td colspan="2">
	<?PHP
	include "banner.php";
	?></td>
 </tr>
 <tr>
  <td width="215" valign="top" nowrap></td>
  <td align="left" valign="top">
	<table width="100%" border="0" cellpadding="0" cellspacing="0">
  <tr> 
   <td valign="top"> 
    <table border=0 cellspacing=0 cellpadding=0 width="100%" height=21> <tr> 
      <td valign=top height=21> 
       <img src="Graphics/intro_tab.gif" width="314" height=21 border=0 align=bottom></td>
      <td valign=top height=21 width="100%"> 
          <table border=0 cellspacing=0 cellpadding=0 width="100%" height="100%">
        <tr> 
         <td valign=top height=20 width="100%"> 
          <p>&nbsp;</p></td>
        </tr>
        <tr> 
         <td valign=top height=1 bgcolor="#7B0101"><img src="Graphics/spacer.gif" width=1 height=1 border=0 align=bottom></td></tr>
       </table></td>
     </tr>
    </table>
    <table border=0 cellspacing=0 cellpadding=0>
     <tr> 
      <td valign=top width=1 bgcolor="#7B0101"><img src="Graphics/spacer.gif" width=1 height=1 align=bottom></td>
      <td valign=top background="Graphics/background_content.gif"> 
       <table border=0 cellspacing=0 cellpadding=10 width="100%">
        <tr> 
          <td valign=top> 
           <p><img src="Graphics/account_header.gif" width="457" height="14"></p>
           <?php if(isset($error))
           			echo "<p>$error</p>";
           		 else
           		 	echo"<p>Please enter your data in the fields below. Choose 'Independent' as the course if you are not part of a taught course. All fields are required to proceed.</p>";
           	?>
          <div align="center">
          <br>
		  <form name="loginform" method="post" action="newuser.php">
           <table border=0 align=center cellpadding=0 cellspacing=4 class="imgborder">
            
            <input type="hidden" id="hiddenwebfirstname" name="hiddenfirstname"/>
            <input type="hidden" id="hiddenweblastname" name="hiddenlastname"/>
            <input type="hidden" id="hiddenwebmailaddress" name="hiddenmailaddress"/>
            <input type="hidden" id="hiddenwebusername" name="hiddenusername"/>
            <input type="hidden" id="hiddenwebinstitution" name="hiddeninstitution"/>
            <input type="hidden" id="hiddenwebcountry" name="hiddencountry"/>
            <input type="hidden" id="hiddenwebbackground" name="hiddenbackground"/>
            <input type="hidden" id="hiddenwebqualifications" name="hiddenqualifications"/>
            <input type="hidden" id="hiddenwebcurrentlevel" name="hiddencurrentlevel"/>
	        <input type="hidden" id="hiddenwebpassword" name="hiddenpassword" value=""/>
	        <input type="hidden" id="hiddenwebhint" name="hiddenhint"/>
	        <input type="hidden" id="hiddenwebcourse" name="hiddencourse"/>
	        <input type="hidden" id="hiddenwebunique" name="hiddenunique"/>
            <tr>
            <td><div align="right" ><b>First name:</b></div></td> 
            <td><input class="formfield" type="text" id="webfirstname" value="<?= $firstname ?>" size="30"></td></tr>
            <tr>
            <td><div align="right" ><b>Last name:</b></div></td> 
            <td><input class="formfield" type="text" id="weblastname" value="<?= $lastname ?>" size="30"></td></tr>
            <tr>
            <td><div align="right" ><b>Email address:</b></div></td> 
            <td><input class="formfield" type="text" id="webmailaddress" value="<?= $mail ?>"size="30"></td></tr>
            <tr>
            <td><div align="right" ><b>Institution/workplace:</b></div></td>
            <td><input class="formfield" type="text" id="webinstitution" value="<?= $institution ?>" size="30"></td></tr>
			<tr>
            <td><div align="right" ><b>Country of student's institution:</b></div></td>
            <td><input class="formfield" type="text" id="webcountry" value="<?= $country ?>" size="30"></td></tr>

            <tr text-align=center>
              <td><div align="right" ><b>Original background:</b></div></td>
              <td><select class="formfield" id="webbackground">
              		<option value="biologist">biologist</option>
              		<option value="biochemist">biochemist</option>
					<option value="chemist">chemist</option>
					<option value="medical">medical</option>
					<option value="pharmacist">pharmacist</option>
					<option value="computer scientist">computer scientist</option>
					<option value="engineer">engineer</option>
					<option value="mathematician">mathematician</option>
					<option value="bioinformatician">bioinformatician</option>
					<option value="physicist">physicist</option>
					<option value="other">other</option>
              </select></td>
            </tr>

			<tr text-align=center>
              <td><div align="right" ><b>Qualifications:</b></div></td>
              <td><select class="formfield" id="webqualifications">
              		<option value="none">none</option>
              		<option value="BSc">BSc</option>
					<option value="MSc">MSc</option>
					<option value="PhDd">PhD</option>
					<option value="other">other</option>
              </select></td>
            </tr>

  <tr> 
   <td><div align="right" ><b>Current Bioinformatics level:</b></div></td>
   <td><select class="formfield" id="webcurrentlevel">
              		<option value="none">none</option>
              		<option value="beginner">beginner</option>
					<option value="competent">competent</option>
					<option value="advanced">advanced</option>
					<option value="expert">expert</option>
              </select></td>
  </tr>
            <tr text-align=center><td><div align="right" ><b>User name:</b></div></td>
            <td>
              <input class="formfield" type="text" id="webusername" value="<?php if(isset($user) && isset($valid_username)) echo $user;?>" size="30"></td></tr>
            <tr text-align=center>
              <td><div align="right" ><b>Password:</b></div></td>
              <td><input class="formfield" type="password" id="webpassword" value="<?= $passwd ?>" size="30"></td>
            </tr>
            <tr text-align=center>
              <td><div align="right" ><b>Confirm password:</b></div></td>
              <td><input class="formfield" type="password" id="webpasswordconfirm" size="30"></td>
            </tr>
            <tr text-align=center>
              <td><div align="right" ><b>Hint to retrieve password:</b></div></td>
              <td><input class="formfield" type="text" id="webhint" value="<?= $hint ?>" size="30"></td>
            </tr>
            <tr text-align=center>
              <td><div align="right" ><b>Choose the course:</b></div></td>
              <td><select class="formfield" id="webcourse"><?= $courseoptions ?></select></td>
            </tr>
            <tr text-align=center>
              <td><div align="right" ><b>Access code for course:</b></div></td>
              <td><input class="formfield" type="password" id="webunique" value="<?= $uniq ?>" size="30"></td>
            </tr>
            </table>     
            
            <br><br>
              <input class="formbutton" onClick="if(!checkNewUser()) return false;submitNewUser('webusername');submitNewUser('webpassword');submitNewUser('webfirstname');submitNewUser('weblastname');submitNewUser('webmailaddress');submitNewUser('webinstitution');submitNewUser('webcountry');submitNewUser('webhint');document.getElementById('hiddenwebbackground').value = document.getElementById('webbackground').value;document.getElementById('hiddenwebqualifications').value = document.getElementById('webqualifications').value;document.getElementById('hiddenwebcurrentlevel').value = document.getElementById('webcurrentlevel').value;document.getElementById('hiddenwebcourse').value = document.getElementById('webcourse').value;submitNewUser('webunique');" type="submit" name="submit" value="Create account">
              <input class="formbutton" type="reset" name="Reset" value="Clear form">
          </form></div>
           <br></td>
        </tr>
       </table></td>
      <td valign=top width=1 bgcolor="#7B0101"> 
       <p><img src="Graphics/spacer.gif" width=1 height=1 align=bottom></p></td>
     </tr>
     <tr> 
      <td valign=top width="1" bgcolor="#7B0101"> 
       <p><img src="Graphics/spacer.gif" width=1 height=1 align=bottom></p></td>
      <td valign=top width="100%" bgcolor="#7B0101"> 
       <p><img src="Graphics/spacer.gif" width=1 height=1 align=bottom></p></td>
      <td valign=top width="1" bgcolor="#7B0101"> 
       <p><img src="Graphics/spacer.gif" width=1 height=1 align=bottom></p></td>
     </tr>
    </table></td>
   <td width=15 valign="top">&nbsp;</td>
  </tr>
 </table></td>
 </tr>
</table>

</div>

<p>&nbsp;</p></body>
</html>

